nexus/test-utils/src/lib.rs

```diff
@@ -260,11 +264,26 @@ pub async fn test_setup_with_config<N: NexusServer>(
     };
     let external_dns_zone_name =
         internal_dns::names::DNS_ZONE_EXTERNAL_TESTING.to_string();
+    let silo_name: Name = "test-suite-silo".parse().unwrap();
+    let user_name = UserId::try_from("test-privileged".to_string()).unwrap();
+    let recovery_silo = RecoverySiloConfig {
+        silo_name: silo_name.clone(),
+        user_name: user_name.clone(),
+        // The test suite's password is "oxide".  This password is only used by
```

Comment:
Suppose we need to change this value, maybe because we're using a new hash function. How would I go about updating this value?

I noticed the string "oxide" hard-coded in some tests, so this seems a little subtle to change.

Reply:
Good question.  What belongs here is a password hash in PHC format using algorithm argon2id and parameters at least as strong as we currently use to generate password hashes for new users.  The API documentation for this type explains this.  It's a little vague about what "secure enough parameters" means because (1) I didn't want to encode that information in multiple places and have it go out of date, (2) it's an internal API anyway, and (3) if you get it wrong, the failure mode is very explicit (e.g., https://github.com/oxidecomputer/omicron/blob/2317079d26e22c78c67054e4ec7fa2a2ad5aeb79/nexus/passwords/src/lib.rs#L548-L549).

As you pointed out, it _also_ needs to be consistent with any places that _use_ the password.  There shouldn't be many of these.  Presumably tests will fail (somewhat explicitly -- with a login failure) if that happens.  There are a few different places that use the same password (e.g., `omicron-dev run-all`), but that's only for convenience.  If you changed this one and "forgot" to update that one, that's fine.

Given these constraints, you can construct this hash with any tool that can generate Argon2 hashes.  I use [argon2.online](https://argon2.online/) for test stuff like this.

Background: I started from the position that it's safer to transfer the password hash in the APIs (from Wicket -> RSS and from RSS -> Nexus) rather than the password itself.  That said, for the test suite and `omicron-dev run-all`, we could define a Rust constant which is the password itself, and then have us hash that and put _that_ into the API calls.  That might require more things outside of Nexus to depend on `nexus-passwords`, but maybe that's okay, and probably has to happen eventually if Wicket is to hash whatever password the user provides.

---

I'll think a bit more about whether to commonize some of the password and hash uses in the test suite, `run-all`, and sim sled agent.  (I would probably _not_ commonize these with the two config-rss.toml files, even though they use the same password, because (1) nothing depends on these being the same as each other or anything else, and (2) these are supposed to be exactly what would come in via RSS, so I don't want to put a cleartext password there, and (3) I don't think there's a way to refer to a Rust constant in those.

Reply:
Updated in the latest commit.  With each hardcoded hash I added a comment explaining what depends on it and referring people to the relevant docs.  I also had the test suite itself encode the password rather than the hash so that it can generate its own hash and we don't need this hardcoded there.